app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntryLogicTest.kt

```diff
@@ -889,6 +890,133 @@ class BloodPressureEntrySheetLogicTest {
     verifyNoMoreInteractions(ui)
   }
 
+  @Test
+  fun `when a different user clicks on save while updating a BP, then the updated BP measurement should be saved with the user's ID and the corresponding facility ID`() {
+    // given
+    reset(userSession, facilityRepository)
+    val userFromDifferentFacility = PatientMocker.loggedInUser(uuid = UUID.fromString("4844b826-a162-49fe-b92c-962da172e86c"))
+    val differentFacility = PatientMocker.facility(uuid = UUID.fromString("f895b54f-ee32-4471-bc0c-a91b80368778"))
+
+    whenever(userSession.requireLoggedInUser()).doReturn(Observable.just(userFromDifferentFacility))
+    whenever(facilityRepository.currentFacility(userFromDifferentFacility)).doReturn(Observable.just(differentFacility))
+
+    val oldCreatedAt = LocalDate.of(1990, 1, 13).toUtcInstant(testUserClock)
+    val patientUuid = UUID.fromString("af92b081-0131-4f91-9c28-98da5737945b")
+    val existingBp = PatientMocker.bp(
+        uuid = patientUuid,
+        userUuid = user.uuid,
+        facilityUuid = facility.uuid,
+        systolic = 9000,
+        diastolic = 8999,
+        createdAt = oldCreatedAt,
+        updatedAt = oldCreatedAt,
+        recordedAt = oldCreatedAt
+    )
+
+    val newInputDate = LocalDate.of(1991, 2, 14)
```

Comment:
We also have another live template, `date`, which will generate `LocalDate.parse("2018-01-01")` which we can then edit for our purposes.

Reply:
Fixed.